cruise-control/src/test/java/com/linkedin/kafka/cruisecontrol/servlet/parameters/ParameterJsonStructureTest.java

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2017 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License"). See License in the project root for license information.
+ */
+package com.linkedin.kafka.cruisecontrol.servlet.parameters;
+
+import com.linkedin.cruisecontrol.servlet.parameters.CruiseControlParameters;
+import io.swagger.v3.oas.models.PathItem;
+import io.swagger.v3.oas.models.Paths;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.parameters.Parameter;
+import io.swagger.v3.parser.OpenAPIV3Parser;
+
+import io.swagger.v3.parser.core.models.ParseOptions;
+import io.swagger.v3.parser.core.models.SwaggerParseResult;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+import java.util.Set;
+import java.util.Map;
+import java.util.HashMap;
+import java.util.TreeSet;
+
+public class ParameterJsonStructureTest {
+  private Map<String, Class> _endpointToClass;
+  OpenAPI _openAPI;
+  /**
+   * Specify endpoints to be tested
+   */
+  @Before
+  public void setupParameterClasses() {
+    _endpointToClass = new HashMap<>();
+    _endpointToClass.put("/kafkacruisecontrol/rebalance", RebalanceParameters.class);
+    _endpointToClass.put("/kafkacruisecontrol/partition_load", PartitionLoadParameters.class);
```

Comment:
It seems that I need additional config overrides for the configure call to succeed. I tried the code you suggested but ran into the following error:
```
java.lang.IllegalArgumentException: HttpServletRequest configuration is missing from the request.

	at com.linkedin.kafka.cruisecontrol.servlet.parameters.AbstractParameters.configure(AbstractParameters.java:95)
	at com.linkedin.kafka.cruisecontrol.servlet.parameters.BootstrapParameters.configure(BootstrapParameters.java:76)
	at com.linkedin.kafka.cruisecontrol.config.KafkaCruiseControlConfig.getConfiguredInstance(KafkaCruiseControlConfig.java:1838)
	at com.linkedin.kafka.cruisecontrol.servlet.parameters.ParameterJsonStructureTest.setupParameterClasses(ParameterJsonStructureTest.java:45)
```
Do you know how I can create the overrides for the test?

Reply:
Ah, I forget `getConfiguredInstance()` will call `object.configure()` after create the object, which will fail here since we are missing required config(`HttpServletRequest` for `CruiseControlParameters` here). 

Looks like the original way  (i.e. `(CruiseControlParameters (_endpointToClass.get(endpoint.getKey()).newInstance())`  at line 54) is the right way to create object.